src/builder.rs

```diff
@@ -674,13 +980,12 @@ impl Builder {
     ///
     /// // Sign Extended Right Shift
     /// assert_eq!(0b0100_0000i8 >> 2, 0b0001_0000);
-    /// assert_eq!(0b1110_0000i8 >> 1, 0b1111_0000);
-    /// assert_eq!(0b1100_0000i8 >> 2, 0b1111_0000);
-    /// ```
+    /// // -32:0b1110_0000  -64:0b1100_0000  -16:0b1111_0000
+    /// assert_eq!(sign_extended_right_shift(-32,1),-16);
```

Comment:
The original code will go wrong during Doc-tests. Have you seen this problem in CI before? 
[Related logs are here](https://travis-ci.org/TheDan64/inkwell/jobs/505316273)

I once took this code out and tested it. It really can't be compiled. Do you have any good suggestions on how to update this code?

```
failures:
---- src/builder.rs - builder::Builder::build_right_shift (line 975) stdout ----
error: literal out of range for i8
  --> src/builder.rs:983:12
   |
10 | assert_eq!(0b1110_0000i8 >> 1, 0b1111_0000);
   |            ^^^^^^^^^^^^^ help: consider using `u8` instead: `0b1110_0000u8`
   |
   = note: #[deny(overflowing_literals)] on by default
   = note: the literal `0b1110_0000i8` (decimal `224`) does not fit into an `i8` and will become `-32i8`
error: literal out of range for i8
  --> src/builder.rs:983:32
   |
10 | assert_eq!(0b1110_0000i8 >> 1, 0b1111_0000);
   |                                ^^^^^^^^^^^
   |
   = note: the literal `0b1111_0000` (decimal `240`) does not fit into an `i8` and will become `-16i8`
   = help: consider using `u8` instead
error: literal out of range for i8
  --> src/builder.rs:984:12
   |
11 | assert_eq!(0b1100_0000i8 >> 2, 0b1111_0000);
   |            ^^^^^^^^^^^^^ help: consider using `u8` instead: `0b1100_0000u8`
   |
   = note: the literal `0b1100_0000i8` (decimal `192`) does not fit into an `i8` and will become `-64i8`
error: literal out of range for i8
  --> src/builder.rs:984:32
   |
11 | assert_eq!(0b1100_0000i8 >> 2, 0b1111_0000);
   |                                ^^^^^^^^^^^
   |
   = note: the literal `0b1111_0000` (decimal `240`) does not fit into an `i8` and will become `-16i8`
   = help: consider using `u8` instead
```

Reply:
This is working on master, so I'm not clear on what has changed...

Reply:
Can you think of any reason why rust wouldn't interpret `0b1110_0000i8` as a negative? There are only 8 bits in an i8 after all

Reply:
Oh, I suppose this just compiled on 2015e, but they made it not compile in 2018e. What a strange breaking change to make.

Reply:
> Oh, I suppose this just compiled on 2015e, but they made it not compile in 2018e. What a strange breaking change to make.

Yeah, I am also very surprised by the error that occurred here.

Reply:
https://github.com/rust-lang/rust/issues/54502#issuecomment-472438189

Reply:
@longwusha What do you think about just taking what we had here and converting them to comments?
For example,
`// 0b1100_0000i8 >> 2 == 0b1111_0000`

Reply:
If you remove the i8 suffix, you should treat it as an unsigned integer by default, so you won't associate the right shift operation with the sign extended right shift. In that case, the calculation results will look wrong or strange.

Reply:
Adding the #[allow(overflowing_literals)] annotation near this may be a better idea. We can add the link to the issue of rust in the documentation to help everyone understand why we do this.

Reply:
I will try it out and see if adding #[allow(overflowing_literals)] annotations around this will solve the problem.